src/preamble.js

```diff
@@ -1714,6 +1714,12 @@ function intArrayFromString(stringy, dontAddNull, length) {
 {{{ maybeExport('intArrayFromString') }}}
 
 function intArrayToString(array) {
+  if (typeof TextDecoder !== 'undefined') {
```

Comment:
Might be best to do it in another PR, then. I seem to recall @juj having a PR for TextDecoder optimizations, not sure what ended up there, we should figure that out.